app/build.gradle.kts

```diff
@@ -5,6 +5,7 @@ plugins {
     id("com.android.application")
     id("com.github.spotbugs")
     id("org.jetbrains.kotlin.android")
+    id("org.jetbrains.kotlin.plugin.compose") version "2.0.0"
```

Comment:
Maybe try updating to the latest kotlin release. There are usually a fair bit of performance improvements and bug fixes for compose.